In [1]:
import pandas as pd

In [2]:
data = pd.read_csv("C:\\Users\\solba\\dacon-project\\data\\raw\\train.csv")

In [3]:
data

ID  generation  school1               major type       major1_1  \
0    TRAIN_000           9       22  복수 전공 ( 다중전공, 이중전공 포함 )          경제통상학   
1    TRAIN_001           9        1  복수 전공 ( 다중전공, 이중전공 포함 )           자연과학   
2    TRAIN_002           9       27                    단일 전공            예체능   
3    TRAIN_003           9        1  복수 전공 ( 다중전공, 이중전공 포함 )           사회과학   
4    TRAIN_004           9       16  복수 전공 ( 다중전공, 이중전공 포함 )  IT(컴퓨터 공학 포함)   
..         ...         ...      ...                      ...            ...   
743  TRAIN_743           9       13  복수 전공 ( 다중전공, 이중전공 포함 )           자연과학   
744  TRAIN_744           9       43                    단일 전공            경영학   
745  TRAIN_745           9       11  복수 전공 ( 다중전공, 이중전공 포함 )            교육학   
746  TRAIN_746           9       30  복수 전공 ( 다중전공, 이중전공 포함 )  IT(컴퓨터 공학 포함)   
747  TRAIN_747           9       25                    단일 전공           자연과학   

          major1_2  major_data  job  class1  class2  ...  \
0             자연과학       False  대학생       1     4.0  ...   
1    IT(컴퓨터 공학 포함)        True  대학생       8     NaN  ...   
2              NaN       False  대학생       7     NaN  ...   
3    IT(컴퓨터 공학 포함)       False  대학생       7     NaN  ...   
4    IT(컴퓨터 공학 포함)        True  대학생       8     NaN  ...   
..             ...         ...  ...     ...     ...  ...   
743  IT(컴퓨터 공학 포함)        True  대학생       7     NaN  ...   
744            NaN        True  대학생      12     NaN  ...   
745           자연과학       False  대학생       2     NaN  ...   
746            NaN       False  대학생       5     NaN  ...   
747            NaN        True  대학생       4     NaN  ...   

      incumbents_company_level  incumbents_lecture_type  \
0                  해외 기업 (빅테크)               온, 오프라인 동시   
1       국내 빅테크 IT 계열 (네카라쿠배당토)                     오프라인   
2    국내 대기업 IT 계열 (금융, 제조 ...)                     오프라인   
3       국내 빅테크 IT 계열 (네카라쿠배당토)               온, 오프라인 동시   
4    국내 대기업 IT 계열 (금융, 제조 ...)               온, 오프라인 동시   
..                         ...                      ...   
743  국내 대기업 IT 계열 (금융, 제조 ...)               온, 오프라인 동시   
744  국내 대기업 IT 계열 (금융, 제조 ...)               온, 오프라인 동시   
745                해외 기업 (빅테크)               온, 오프라인 동시   
746  국내 대기업 IT 계열 (금융, 제조 ...)               온, 오프라인 동시   
747  국내 대기업 IT 계열 (금융, 제조 ...)                      온라인   

      incumbents_lecture_scale  \
0    100명 이상의 리스너와 10명 이상의 현직자   
1    3~50명 내외의 강의 리스너와 1명의 현직자   
2    3~50명 내외의 강의 리스너와 1명의 현직자   
3    3~50명 내외의 강의 리스너와 1명의 현직자   
4    3~50명 내외의 강의 리스너와 1명의 현직자   
..                         ...   
743  3~50명 내외의 강의 리스너와 1명의 현직자   
744  100명 이상의 리스너와 10명 이상의 현직자   
745  100명 이상의 리스너와 10명 이상의 현직자   
746  3~50명 내외의 강의 리스너와 1명의 현직자   
747  3~50명 내외의 강의 리스너와 1명의 현직자   

                       incumbents_lecture_scale_reason interested_company  \
0                         다양한 사람들과 만나서 생각을 교류할 수 있기 때문   구글 딥마인드, 카카오 브레인   
1     더 많은 사람들이 있으면 제가 예상하지 못한 질문도 할 수 있다고 생각하기 때문입니다.              제일 기획   
2    인원이 너무 적으면 서로 부담스러울 수 있을 것 같지만 너무 많으면 너무 피상적인 ...               Lg전자   
3                            너무 많은 인원이 강의하면 루즈해질 것 같아서                네이버   
4                             다양한 사람들에게 기회가 있으면 좋겠습니다.                네이버   
..                                                 ...                ...   
743                조금이라도 많은 사람이 있어야 다양한 질문들을 접해볼 수 있어서        삼성, LG, 금감원   
744                                많은 현직자들의 이야기를 듣고 싶다                 현대   
745                             인원이 많으면 접근성이 용이할 것 같아서                네이버   
746                                             집중하기위해          찾아보는중입니다.   
747                                               집중도                한국은행   

                                       expected_domain contest_participation  \
0                                  M. 전문, 과학 및 기술 서비스업                   NaN   
1                     J. 정보통신업, O. 공공 행정, 국방 및 사회보장 행정                   NaN   
2          C. 제조업, K. 금융 및 보험업, R. 예술, 스포츠 및 여가관련 서비스업     

In [4]:
data.columns

Index(['ID', 'generation', 'school1', 'major type', 'major1_1', 'major1_2',
       'major_data', 'job', 'class1', 'class2', 'class3', 'class4',
       're_registration', 'contest_award', 'nationality', 'inflow_route',
       'whyBDA', 'what_to_gain', 'hope_for_group', 'previous_class_3',
       'previous_class_4', 'previous_class_5', 'previous_class_6',
       'previous_class_7', 'previous_class_8', 'major_field',
       'desired_career_path', 'completed_semester', 'project_type',
       'time_input', 'desired_job', 'certificate_acquisition',
       'desired_certificate', 'desired_job_except_data', 'incumbents_level',
       'incumbents_lecture', 'incumbents_company_level',
       'incumbents_lecture_type', 'incumbents_lecture_scale',
       'incumbents_lecture_scale_reason', 'interested_company',
       'expected_domain', 'contest_participation', 'idea_contest',
       'onedayclass_topic', 'completed'],
      dtype='object')

In [5]:
# 칼럼 설명
column_descriptions = {
    'ID': '학생 ID (예: TRAIN_000, TRAIN_001)',
    'generation': '기수 (9기 등)',
    'school1': '학교 코드',
    'major type': '전공 유형 (단일 전공, 복수 전공 등)',
    'major1_1': '주 전공 1',
    'major1_2': '주 전공 2',
    'major_data': '데이터 관련 전공 여부 (True/False)',
    'job': '직업 (대학생, 직장인 등)',
    'class1': '수강 클래스 1',
    'class2': '수강 클래스 2',
    'class3': '수강 클래스 3',
    'class4': '수강 클래스 4',
    're_registration': '재등록 여부',
    'contest_award': '공모전 수상 여부',
    'nationality': '국적 (내국인, 외국인 등)',
    'inflow_route': '유입 경로 (에브리타임, 지인 추천 등)',
    'whyBDA': 'BDA 참가 이유',
    'what_to_gain': 'BDA를 통해 얻고 싶은 것',
    'hope_for_group': '그룹/강의 운영 방식에 대한 희망사항',
    'previous_class_3': '이전 3기 수강 여부',
    'previous_class_4': '이전 4기 수강 여부',
    'previous_class_5': '이전 5기 수강 여부',
    'previous_class_6': '이전 6기 수강 여부',
    'previous_class_7': '이전 7기 수강 여부',
    'previous_class_8': '이전 8기 수강 여부',
    'major_field': '관심 전공 분야',
    'desired_career_path': '희망 경력 경로 (취업, 대학원 진학 등)',
    'completed_semester': '이수한 학기 수',
    'project_type': '프로젝트 유형 (팀 프로젝트, 개인 프로젝트)',
    'time_input': '투자 시간',
    'desired_job': '희망 직무',
    'certificate_acquisition': '취득한 자격증',
    'desired_certificate': '희망 자격증',
    'desired_job_except_data': '데이터 분야 외 희망 직무',
    'incumbents_level': '현직자 경력 수준 (주니어, 시니어 등)',
    'incumbents_lecture': '현직자 강의 주제',
    'incumbents_company_level': '현직자 소속 기업 규모',
    'incumbents_lecture_type': '현직자 강의 유형',
    'incumbents_lecture_scale': '희망 강의 규모',
    'incumbents_lecture_scale_reason': '강의 규모 선택 이유',
    'interested_company': '관심 기업',
    'expected_domain': '희망 산업 분야',
    'contest_participation': '공모전 참여 여부',
    'idea_contest': '아이디어 공모전 참여',
    'onedayclass_topic': '원데이클래스 주제',
    'completed': '과정 이수 완료 여부 (0: 미완료, 1: 완료)'
}

print("="*80)
print("칼럼 설명 (총 43개)")
print("="*80)
for i, (col, desc) in enumerate(column_descriptions.items(), 1):
    print(f"{i:2d}. {col:30s} - {desc}")
print("="*80)

칼럼 설명 (총 43개)
 1. ID                             - 학생 ID (예: TRAIN_000, TRAIN_001)
 2. generation                     - 기수 (9기 등)
 3. school1                        - 학교 코드
 4. major type                     - 전공 유형 (단일 전공, 복수 전공 등)
 5. major1_1                       - 주 전공 1
 6. major1_2                       - 주 전공 2
 7. major_data                     - 데이터 관련 전공 여부 (True/False)
 8. job                            - 직업 (대학생, 직장인 등)
 9. class1                         - 수강 클래스 1
10. class2                         - 수강 클래스 2
11. class3                         - 수강 클래스 3
12. class4                         - 수강 클래스 4
13. re_registration                - 재등록 여부
14. contest_award                  - 공모전 수상 여부
15. nationality                    - 국적 (내국인, 외국인 등)
16. inflow_route                   - 유입 경로 (에브리타임, 지인 추천 등)
17. whyBDA                         - BDA 참가 이유
18. what_to_gain                   - BDA를 통해 얻고 싶은 것
19. hope_for_group                 - 그룹/강의 운영 방식에 대한 희망사항
20. previous_clas

In [6]:
import numpy as np

# 각 열의 unique 값 분석
print("="*100)
print("각 열의 데이터 타입 및 Unique 값 분석")
print("="*100)

for col in data.columns:
    unique_count = data[col].nunique()
    null_count = data[col].isnull().sum()
    dtype = data[col].dtype
    
    print(f"\n📊 {col}")
    print(f"   데이터 타입: {dtype} | Unique 값 개수: {unique_count} | Null 개수: {null_count}")
    
    # unique 값이 20개 이하면 모두 출력, 초과면 샘플 출력
    unique_values = data[col].dropna().unique()
    
    if unique_count <= 20:
        print(f"   값: {sorted(unique_values) if dtype in ['int64', 'float64'] else sorted(unique_values)}")
    else:
        print(f"   샘플 값: {list(unique_values[:10])}")
        print(f"   ... (총 {unique_count}개의 unique 값)")

print("\n" + "="*100)

각 열의 데이터 타입 및 Unique 값 분석

📊 ID
   데이터 타입: object | Unique 값 개수: 748 | Null 개수: 0
   샘플 값: ['TRAIN_000', 'TRAIN_001', 'TRAIN_002', 'TRAIN_003', 'TRAIN_004', 'TRAIN_005', 'TRAIN_006', 'TRAIN_007', 'TRAIN_008', 'TRAIN_009']
   ... (총 748개의 unique 값)

📊 generation
   데이터 타입: int64 | Unique 값 개수: 1 | Null 개수: 0
   값: [np.int64(9)]

📊 school1
   데이터 타입: int64 | Unique 값 개수: 92 | Null 개수: 0
   샘플 값: [np.int64(22), np.int64(1), np.int64(27), np.int64(16), np.int64(81), np.int64(20), np.int64(28), np.int64(40), np.int64(18), np.int64(5)]
   ... (총 92개의 unique 값)

📊 major type
   데이터 타입: object | Unique 값 개수: 3 | Null 개수: 22
   값: ['단일 전공', '단일 전공공학 (컴퓨터 공학 제외)', '복수 전공 ( 다중전공, 이중전공 포함 )']

📊 major1_1
   데이터 타입: object | Unique 값 개수: 11 | Null 개수: 20
   값: ['IT(컴퓨터 공학 포함)', '경영학', '경제통상학', '교육학', '기타', '법학', '사회과학', '예체능', '의약학', '인문학', '자연과학']

📊 major1_2
   데이터 타입: object | Unique 값 개수: 10 | Null 개수: 439
   값: ['IT(컴퓨터 공학 포함)', '경영학', '경제통상학', '교육학', '기타', '사회과학', '예체능', '의약학', '인문학', '자연과학']

====================================================================================================
각 열의 데이터 타입 및 Unique 값 분석
====================================================================================================

📊 ID
   데이터 타입: object | Unique 값 개수: 748 | Null 개수: 0
   샘플 값: ['TRAIN_000', 'TRAIN_001', 'TRAIN_002', 'TRAIN_003', 'TRAIN_004', 'TRAIN_005', 'TRAIN_006', 'TRAIN_007', 'TRAIN_008', 'TRAIN_009']
   ... (총 748개의 unique 값)

📊 generation
   데이터 타입: int64 | Unique 값 개수: 1 | Null 개수: 0
   값: [np.int64(9)]

📊 school1
   데이터 타입: int64 | Unique 값 개수: 92 | Null 개수: 0
   샘플 값: [np.int64(22), np.int64(1), np.int64(27), np.int64(16), np.int64(81), np.int64(20), np.int64(28), np.int64(40), np.int64(18), np.int64(5)]
   ... (총 92개의 unique 값)

📊 major type
   데이터 타입: object | Unique 값 개수: 3 | Null 개수: 22
   값: ['단일 전공', '단일 전공공학 (컴퓨터 공학 제외)', '복수 전공 ( 다중전공, 이중전공 포함 )']

📊 major1_1
   데이터 타입: object | Unique 값 개수: 11 | Null 개수: 20
   값: ['IT(컴퓨터 공학 포함)', '경영학', '경제통상학', '교육학', '기타', '법학', '사회과학', '예체능', '의약학', '인문학', '자연과학']

📊 major1_2
   데이터 타입: object | Unique 값 개수: 10 | Null 개수: 439
   값: ['IT(컴퓨터 공학 포함)', '경영학', '경제통상학', '교육학', '기타', '사회과학', '예체능', '의약학', '인문학', '자연과학']

📊 major_data
   데이터 타입: bool | Unique 값 개수: 2 | Null 개수: 0
   값: [np.False_, np.True_]

📊 job
   데이터 타입: object | Unique 값 개수: 4 | Null 개수: 0
   값: ['대학생', '대학원생', '직장인', '취준생']

📊 class1
   데이터 타입: int64 | Unique 값 개수: 10 | Null 개수: 0
   값: [np.int64(1), np.int64(2), np.int64(4), np.int64(5), np.int64(6), np.int64(7), np.int64(8), np.int64(11), np.int64(12), np.int64(13)]

📊 class2
   데이터 타입: float64 | Unique 값 개수: 9 | Null 개수: 579
   값: [np.float64(2.0), np.float64(4.0), np.float64(5.0), np.float64(6.0), np.float64(7.0), np.float64(8.0), np.float64(11.0), np.float64(12.0), np.float64(13.0)]

📊 class3
   데이터 타입: float64 | Unique 값 개수: 5 | Null 개수: 734
   값: [np.float64(5.0), np.float64(6.0), np.float64(8.0), np.float64(12.0), np.float64(13.0)]

📊 class4
   데이터 타입: float64 | Unique 값 개수: 1 | Null 개수: 747
   값: [np.float64(12.0)]

📊 re_registration
   데이터 타입: object | Unique 값 개수: 2 | Null 개수: 0
   값: ['아니요', '예']

📊 contest_award
   데이터 타입: float64 | Unique 값 개수: 0 | Null 개수: 748
   값: []

📊 nationality
   데이터 타입: object | Unique 값 개수: 2 | Null 개수: 1
   값: ['내국인', '외국인']

📊 inflow_route
   데이터 타입: object | Unique 값 개수: 8 | Null 개수: 0
   값: ['교내 플랫폼 (에브리타임 X)', '기존 학회원 또는 운영진', '기타', '대외활동 사이트(링커리어, 캠퍼스픽, 캠퍼즈, 위비티 등)', '에브리타임', '인스타그램', '인터넷 검색', '지인 추천']

📊 whyBDA
   데이터 타입: object | Unique 값 개수: 7 | Null 개수: 0
   값: ['BDA 학회원만의 혜택을 누리고 싶어서(현직자 강연, 잡 페스티벌, 기업연계 공모전 등)', '시간적으로 부담이 없어서', '이전 기수에 매우 만족해서', '큰 규모인 만큼, 커리큘럼이나 운영 등 관리가 잘 될것 같아서', '학회 가입 시 코딩 테스트, 면접 등을 보지 않아서', '현직자의 강의를 듣고 싶어서', '혼자 공부하기 어려워서']

📊 what_to_gain
   데이터 타입: object | Unique 값 개수: 9 | Null 개수: 0
   값: ['공모전 경험', '공모전과 프로젝트', '기초적인 python, sql 지식 숙지', '데이터 분석 역량', '분석 역량, 공모전 경험, 인적 네트워크', '사실 한가지를 고르기 어렵습니다..ㅠㅠ 학회를 통해 위의 4가지와 협업에 계신 전문가님들에게 유익한 정보를 얻고 소중한 경험을 하고 싶습니다.', '위 항목 전체', '인적 네트워크', '프로젝트 경험']

📊 hope_for_group
   데이터 타입: object | Unique 값 개수: 3 | Null 개수: 0
   값: ['네. 오프라인으로 참여하고 싶어요', '네. 온라인으로 참여하고 싶어요', '아니요. 개인적으로 학회 활동을 하고 싶어요']

📊 previous_class_3
   데이터 타입: object | Unique 값 개수: 3 | Null 개수: 602
   값: ['0005 : 데이터 분석 입문반 (통계)', '0006 : 데이터 분석 전처리반(판다스)', '해당없음']

📊 previous_class_4
   데이터 타입: object | Unique 값 개수: 4 | Null 개수: 602
   값: ['0001 : 파이썬 문법 기초반', '0005 : 데이터 분석 입문반 (통계)', '0009 : 데이터 분석 개념반', '해당없음']

📊 previous_class_5
   데이터 타입: object | Unique 값 개수: 5 | Null 개수: 602
   값: ['0001 : 파이썬 문법 기초반', '0004 : 데이터 분석 입문반 (머신러닝)', '0005 : 데이터 분석 입문반 (통계)', '0006 : 데이터 분석 전처리반(판다스)', '해당없음']

📊 previous_class_6
   데이터 타입: object | Unique 값 개수: 10 | Null 개수: 602
   값: ['0001 : 파이썬 문법 기초반', '0001 : 파이썬 문법 기초반, 0003 : R문법 기초반', '0001 : 파이썬 문법 기초반, 0004 : 데이터 분석 입문반 (머신러닝)', '0004 : 데이터 분석 입문반 (머신러닝)', '0005 : 데이터 분석 입문반 (통계)', '0006 : 데이터 분석 전처리반(판다스)', '0007 : 데이터 분석 전처리 적용반', '0008 : 데이터 분석 모델링반 (ML1)', '0009 : 데이터 분석 개념반', '해당없음']

📊 previous_class_7
   데이터 타입: object | Unique 값 개수: 12 | Null 개수: 602
   값: ['0001 : 파이썬 문법 기초반', '0001 : 파이썬 문법 기초반, 0002 : SQL 문법 기초반', '0001 : 파이썬 문법 기초반, 0009 : 데이터 분석 개념반', '0002 : SQL 문법 기초반', '0002 : SQL 문법 기초반, 0004 : 데이터 분석 입문반 (머신러닝)', '0004 : 데이터 분석 입문반 (머신러닝)', '0005 : 데이터 분석 입문반 (통계)', '0006 : 데이터 분석 전처리반(판다스)', '0008 : 데이터 분석 모델링반 (ML1)', '0009 : 데이터 분석 개념반', '해당없음', '해당없음, 0004 : 데이터 분석 입문반 (머신러닝)']

📊 previous_class_8
   데이터 타입: object | Unique 값 개수: 16 | Null 개수: 602
   값: ['0001 : 파이썬 문법 기초반', '0001 : 파이썬 문법 기초반, 0002 : SQL 문법 기초반', '0001 : 파이썬 문법 기초반, 0004 : 데이터 분석 입문반 (머신러닝)', '0001 : 파이썬 문법 기초반, 0009 : 데이터 분석 개념반', '0002 : SQL 문법 기초반', '0002 : SQL 문법 기초반, 0005 : 데이터 분석 입문반 (통계)', '0002 : SQL 문법 기초반, 0006 : 데이터 분석 전처리반(판다스)', '0004 : 데이터 분석 입문반 (머신러닝)', '0004 : 데이터 분석 입문반 (머신러닝), 0005 : 데이터 분석 입문반 (통계)', '0004 : 데이터 분석 입문반 (머신러닝), 0006 : 데이터 분석 전처리반(판다스)', '0005 : 데이터 분석 입문반 (통계)', '0005 : 데이터 분석 입문반 (통계), 0006 : 데이터 분석 전처리반(판다스)', '0006 : 데이터 분석 전처리반(판다스)', '0007 : 데이터 분석 전처리 적용반', '0008 : 데이터 분석 모델링반 (ML1)', '해당없음']

📊 major_field
   데이터 타입: object | Unique 값 개수: 43 | Null 개수: 23
   샘플 값: ['자연과학, 사회과학', '사회과학', '예체능', '법학', 'IT (컴퓨터 공학 포함), 인문학', '경영학', 'IT (컴퓨터 공학 포함), 사회과학', '경영학, 경제통상학', '공학 (컴퓨터 공학 제외)', 'IT (컴퓨터 공학 포함)']
   ... (총 43개의 unique 값)

📊 desired_career_path
   데이터 타입: object | Unique 값 개수: 4 | Null 개수: 0
   값: ['대학원 진학', '이직', '창업', '취업']

📊 completed_semester
   데이터 타입: float64 | Unique 값 개수: 13 | Null 개수: 28
   값: [np.float64(0.0), np.float64(1.0), np.float64(2.0), np.float64(3.0), np.float64(4.0), np.float64(5.0), np.float64(6.0), np.float64(7.0), np.float64(8.0), np.float64(9.0), np.float64(10.0), np.float64(2020.02), np.float64(20241.0)]

📊 project_type
   데이터 타입: object | Unique 값 개수: 2 | Null 개수: 0
   값: ['개인', '팀']

📊 time_input
   데이터 타입: float64 | Unique 값 개수: 14 | Null 개수: 0
   값: [np.float64(0.0), np.float64(0.7), np.float64(1.0), np.float64(1.5), np.float64(2.0), np.float64(2.5), np.float64(3.0), np.float64(4.0), np.float64(5.0), np.float64(6.0), np.float64(7.0), np.float64(8.0), np.float64(10.0), np.float64(24.0)]

📊 desired_job
   데이터 타입: object | Unique 값 개수: 174 | Null 개수: 0
   샘플 값: ['C. 데이터 사이언티스트', 'B. 데이터 분석가, C. 데이터 사이언티스트', 'B. 데이터 분석가, C. 데이터 사이언티스트, I. 마케터', 'B. 데이터 분석가', 'B. 데이터 분석가, F. UI/UX 디자이너, G. PM/서비스 기획자, I. 마케터, J. MD', 'A. 데이터 엔지니어, B. 데이터 분석가', 'B. 데이터 분석가, C. 데이터 사이언티스트, I. 마케터, J. MD', 'G. PM/서비스 기획자, I. 마케터', 'A. 데이터 엔지니어, B. 데이터 분석가, E. 소프트웨어 개발자', 'A. 데이터 엔지니어, B. 데이터 분석가, C. 데이터 사이언티스트, D. 인공지능 전문가']
   ... (총 174개의 unique 값)

📊 certificate_acquisition
   데이터 타입: object | Unique 값 개수: 27 | Null 개수: 0
   샘플 값: ['컴퓨터활용능력', '없음', 'ADsP', 'SQLD, 정보처리기사', '구글 애널리스트', 'SQLD, 구글 애널리스트', 'SQLD', 'ADsP, SQLD', '정보처리기사', '준비중: SQLD']
   ... (총 27개의 unique 값)

📊 desired_certificate
   데이터 타입: object | Unique 값 개수: 91 | Null 개수: 0
   샘플 값: ['ADsP, SQLD, 빅데이터 분석 기사, 태블로 관련 자격증, 구글 애널리스트', 'ADsP, SQLD, 빅데이터 분석 기사, 정보처리기사', 'SQLD, 빅데이터 분석 기사, 정보처리기사', 'ADsP, 빅데이터 분석 기사', 'SQLD, 정보처리기사, 태블로 관련 자격증, 구글 애널리스트', 'ADsP, SQLD, 빅데이터 분석 기사', 'ADsP, SQLD', 'ADsP, SQLD, 빅데이터 분석 기사, 정보처리기사, 클라우드 관련 자격증', '빅데이터 분석 기사, 정보처리기사', '빅데이터 분석 기사']
   ... (총 91개의 unique 값)

📊 desired_job_except_data
   데이터 타입: object | Unique 값 개수: 99 | Null 개수: 0
   샘플 값: ['A. 금융 / 보험 직무', 'D. 소프트웨어 개발자, F. PM / 서비스 기획자', 'A. 금융 / 보험 직무, B. 기획 / 전략 / 경영 직무, H. 마케터 / 영업관리', 'A. 금융 / 보험 직무, B. 기획 / 전략 / 경영 직무, F. PM / 서비스 기획자', 'B. 기획 / 전략 / 경영 직무, E. UI/UX 디자이너, F. PM / 서비스 기획자, H. 마케터 / 영업관리', 'B. 기획 / 전략 / 경영 직무, H. 마케터 / 영업관리', 'A. 금융 / 보험 직무, B. 기획 / 전략 / 경영 직무, C. MD, F. PM / 서비스 기획자', 'B. 기획 / 전략 / 경영 직무, C. MD, F. PM / 서비스 기획자, H. 마케터 / 영업관리', 'D. 소프트웨어 개발자', 'A. 금융 / 보험 직무, B. 기획 / 전략 / 경영 직무, F. PM / 서비스 기획자, H. 마케터 / 영업관리']
   ... (총 99개의 unique 값)

📊 incumbents_level
   데이터 타입: object | Unique 값 개수: 2 | Null 개수: 0
   값: ['시니어 (10년차 ~)', '주니어 (0~3년차)']

📊 incumbents_lecture
   데이터 타입: object | Unique 값 개수: 10 | Null 개수: 0
   값: ['산업 트렌드 (예시: ChatGPT로 인한 직무 변화)', '원동력', '위 3개 전부 다', '위 항목 전부 다', '위에 3가지 모두 듣고 싶습니다', '직무 강의 (예시: 실무 진행 방식 및 직무 준비생을 위한 팁)', '취업하기 위해 준비했던 각종 데이터 관련 경험들', '커리어 패스 과정 (예시: 비전공자/전공자의 취업 준비 및 이직 과정)', '커리어 패스 과정과 직무 강의 둘다 궁금합니다 ', '커리어패스, 산업 트렌드']

📊 incumbents_company_level
   데이터 타입: object | Unique 값 개수: 17 | Null 개수: 0
   값: ['...', '1순위: 국내 대기업 IT 계열. 2순위: 해외 기업(빅테크)', '게임회사', '공간정보 관련 직무 계열', '국내 대기업 IT 계열 (금융, 제조 ...)', '국내 대기업 IT 계열과 해외 기업 현직자에게 가장 듣고 싶습니다', '국내 빅테크 IT 계열 (네카라쿠배당토)', '국내 빅테크 IT계열, 해외 기업 둘 다', '국내와 해외 기업 현직자 모두에게 들었으면 좋겠다.', '국내외 반도체 기업', '바이오인공지능 계열', '배터리 관련 기업', '스타트업 C-level', '스포츠구단', '은행 및 보험 쪽 IT', '잘 모르겠음', '해외 기업 (빅테크)']

📊 incumbents_lecture_type
   데이터 타입: object | Unique 값 개수: 3 | Null 개수: 0
   값: ['오프라인', '온, 오프라인 동시', '온라인']

📊 incumbents_lecture_scale
   데이터 타입: object | Unique 값 개수: 6 | Null 개수: 0
   값: ['...', '100명 이상의 리스너와 1-2명의 현직자', '100명 이상의 리스너와 10명 이상의 현직자', '100명 이상의 리스너와 3명의 현직자', '10명 내외의 강의 리스너와 1명의 현직자', '3~50명 내외의 강의 리스너와 1명의 현직자']

📊 incumbents_lecture_scale_reason
   데이터 타입: object | Unique 값 개수: 691 | Null 개수: 0
   샘플 값: ['다양한 사람들과 만나서 생각을 교류할 수 있기 때문', '더 많은 사람들이 있으면 제가 예상하지 못한 질문도 할 수 있다고 생각하기 때문입니다.', '인원이 너무 적으면 서로 부담스러울 수 있을 것 같지만 너무 많으면 너무 피상적인 강의가 될 것 같아서', '너무 많은 인원이 강의하면 루즈해질 것 같아서', '다양한 사람들에게 기회가 있으면 좋겠습니다.', '그래야 직접 들을 확률이 높아져서', '인원이 너무 많으면 집중하기 어렵기 때문이다', '더 집중해서 강의를 들을 수 있고 세부적인 질문도 가능할 것 같아서 10명 내외로 선택하였습니다.', '리스너들과도 적당히 소통할 수 있으면서 현직자의 강의에 집중할 수 있어서', '인원이 적어야 프로젝트성 그룹 같은경우는 집중할수있을것같다']
   ... (총 691개의 unique 값)

📊 interested_company
   데이터 타입: object | Unique 값 개수: 438 | Null 개수: 0
   샘플 값: ['구글 딥마인드, 카카오 브레인', '제일 기획', 'Lg전자', '네이버', '삼성', '아직 없음', '쏘카', '아모레퍼시픽', '아이디어스', '인바이테']
   ... (총 438개의 unique 값)

📊 expected_domain
   데이터 타입: object | Unique 값 개수: 170 | Null 개수: 0
   샘플 값: ['M. 전문, 과학 및 기술 서비스업', 'J. 정보통신업, O. 공공 행정, 국방 및 사회보장 행정', 'C. 제조업, K. 금융 및 보험업, R. 예술, 스포츠 및 여가관련 서비스업', 'J. 정보통신업, K. 금융 및 보험업', 'K. 금융 및 보험업, M. 전문, 과학 및 기술 서비스업, R. 예술, 스포츠 및 여가관련 서비스업', 'K. 금융 및 보험업', 'J. 정보통신업, K. 금융 및 보험업, P. 교육 서비스업, U. 국제 및 외국기관', 'C. 제조업, J. 정보통신업, K. 금융 및 보험업', 'J. 정보통신업', 'K. 금융 및 보험업, M. 전문, 과학 및 기술 서비스업, N. 사업시설 관리, 사업 지원 및 임대 서비스업']
   ... (총 170개의 unique 값)

📊 contest_participation
   데이터 타입: object | Unique 값 개수: 6 | Null 개수: 742
   값: ['1', "국민여행조사' <- 관광지식정보시스템(문화체육관광부)에서 2022 ~ 2023년 국민여행조사 데이터를 수집.", '그 후 네이버 리뷰에서 관광지 리뷰를 전처리 하여(KoNLPy를 사용) 리뷰 데이터로 명소의 특징이 남아있는 명사 데이터를 제하고 모두 제거. 남아있는 변수들중 빈도가 높은 걸로 각각 7개씩 추출후 군집분석 결과와 매칭하여 여행 코스를 짬.', '대상을 50세 이상의 시니어로 선정.', '데이터 :', '전처리 과정 :']

📊 idea_contest
   데이터 타입: float64 | Unique 값 개수: 0 | Null 개수: 748
   값: []

📊 onedayclass_topic
   데이터 타입: object | Unique 값 개수: 41 | Null 개수: 0
   샘플 값: ['Python 응용, 데이터 시각화 (Matplotlib, Seaborn 등), 머신러닝 / 딥러닝 응용, SQL 응용', '머신러닝 / 딥러닝 응용', '머신러닝 / 딥러닝 응용, SQL 응용, 웹 크롤링', 'Python 응용', 'Python 응용, 데이터 시각화 (Matplotlib, Seaborn 등)', 'Python 응용, SQL 응용, 웹 크롤링', 'Python 응용, 데이터 시각화 (Matplotlib, Seaborn 등), 머신러닝 / 딥러닝 응용', 'SQL 응용', '데이터 시각화 (Matplotlib, Seaborn 등), 웹 크롤링', '머신러닝 / 딥러닝 응용, SQL 응용']
   ... (총 41개의 unique 값)

📊 completed
   데이터 타입: int64 | Unique 값 개수: 2 | Null 개수: 0
   값: [np.int64(0), np.int64(1)]

====================================================================================================

# 피처 엔지니어링

In [7]:
# 데이터 재로드 (피처 엔지니어링용)
df_train = pd.read_csv("C:\\Users\\solba\\dacon-project\\data\\raw\\train.csv")
df_test = pd.read_csv("C:\\Users\\solba\\dacon-project\\data\\raw\\test.csv")

print("=" * 100)
print("피처 엔지니어링")
print("=" * 100)

# 불필요한 열 삭제 (generation, contest_award, idea_contest)
df_train = df_train.drop(columns=['generation', 'contest_award', 'idea_contest'])
df_test = df_test.drop(columns=['generation', 'contest_award', 'idea_contest'])

print("✅ 불필요한 열 삭제 (generation, contest_award, idea_contest)")

# 1. major_count: major1_1, major1_2 개수 (0, 1, 2)
df_train['major_count'] = df_train[['major1_1', 'major1_2']].notna().sum(axis=1)
df_test['major_count'] = df_test[['major1_1', 'major1_2']].notna().sum(axis=1)

# 2. is_IT: IT 여부
def check_it(row):
    m1 = str(row['major1_1']) if pd.notna(row['major1_1']) else ""
    m2 = str(row['major1_2']) if pd.notna(row['major1_2']) else ""
    return "IT(컴퓨터 공학 포함)" in m1 or "IT(컴퓨터 공학 포함)" in m2

df_train['is_IT'] = df_train.apply(check_it, axis=1)
df_test['is_IT'] = df_test.apply(check_it, axis=1)

print("✅ major_count, is_IT 생성")

# 3. class_count: 수강 수업 개수
df_train['class_count'] = df_train[['class1', 'class2', 'class3', 'class4']].notna().sum(axis=1)
df_test['class_count'] = df_test[['class1', 'class2', 'class3', 'class4']].notna().sum(axis=1)

print("✅ class_count 생성")

# 4. previous_class_count: 이전 기수 수강 (중복 제거)
def count_prev_classes(row):
    unique_classes = set()
    for col in ['previous_class_3', 'previous_class_4', 'previous_class_5', 
                'previous_class_6', 'previous_class_7', 'previous_class_8']:
        val = row[col]
        if pd.notna(val) and val != "해당없음":
            classes = [c.strip() for c in str(val).split(',')]
            unique_classes.update(classes)
    return len(unique_classes)

df_train['previous_class_count'] = df_train.apply(count_prev_classes, axis=1)
df_test['previous_class_count'] = df_test.apply(count_prev_classes, axis=1)

print("✅ previous_class_count 생성")

# 5. incumbents_lecture_scale_reason을 글자수로 변환
df_train['incumbents_lecture_scale_reason_len'] = df_train['incumbents_lecture_scale_reason'].apply(lambda x: len(str(x)) if pd.notna(x) else 0)
df_test['incumbents_lecture_scale_reason_len'] = df_test['incumbents_lecture_scale_reason'].apply(lambda x: len(str(x)) if pd.notna(x) else 0)

print("✅ incumbents_lecture_scale_reason_len 생성")

피처 엔지니어링
✅ 불필요한 열 삭제 (generation, contest_award, idea_contest)
✅ major_count, is_IT 생성
✅ class_count 생성
✅ previous_class_count 생성
✅ incumbents_lecture_scale_reason_len 생성


In [8]:
# 6~10. 텍스트 나열형 변수를 개수로 변환

def count_items(val):
    """쉼표 단위로 항목 개수 세기. '없음'이거나 NaN이면 0"""
    if pd.isna(val):
        return 0
    val = str(val).strip()
    if val == "없음":
        return 0
    return len([x.strip() for x in val.split(',') if x.strip()])

# desired_job_count
df_train['desired_job_count'] = df_train['desired_job'].apply(count_items)
df_test['desired_job_count'] = df_test['desired_job'].apply(count_items)

# certificate_count ('없음' 처리)
df_train['certificate_count'] = df_train['certificate_acquisition'].apply(count_items)
df_test['certificate_count'] = df_test['certificate_acquisition'].apply(count_items)

# desired_certificate_count
df_train['desired_certificate_count'] = df_train['desired_certificate'].apply(count_items)
df_test['desired_certificate_count'] = df_test['desired_certificate'].apply(count_items)

# desired_job_except_data_count
df_train['desired_job_except_data_count'] = df_train['desired_job_except_data'].apply(count_items)
df_test['desired_job_except_data_count'] = df_test['desired_job_except_data'].apply(count_items)

# onedayclass_topic_count
df_train['onedayclass_topic_count'] = df_train['onedayclass_topic'].apply(count_items)
df_test['onedayclass_topic_count'] = df_test['onedayclass_topic'].apply(count_items)

print("✅ desired_job_count, certificate_count, desired_certificate_count, desired_job_except_data_count, onedayclass_topic_count 생성")

✅ desired_job_count, certificate_count, desired_certificate_count, desired_job_except_data_count, onedayclass_topic_count 생성


In [9]:
# 원본 컬럼 삭제
drop_cols = ['major type', 'major1_1', 'major1_2', 'major_data',
             'class1', 'class2', 'class3', 'class4',
             'previous_class_3', 'previous_class_4', 'previous_class_5', 
             'previous_class_6', 'previous_class_7', 'previous_class_8',
             'desired_job', 'certificate_acquisition', 'desired_certificate', 
             'desired_job_except_data', 'onedayclass_topic',
             'incumbents_lecture_scale_reason', 'interested_company', 
             'expected_domain', 'completed_semester', 'major_field']

df_train = df_train.drop(columns=drop_cols)
df_test = df_test.drop(columns=drop_cols)

print("✅ 원본 컬럼 삭제 완료\n")

# 결과 확인
print("=" * 100)
print("최종 컬럼 목록:")
print("=" * 100)
print(df_train.columns.tolist())
print(f"\nTrain shape: {df_train.shape}")
print(f"Test shape: {df_test.shape}")

✅ 원본 컬럼 삭제 완료

최종 컬럼 목록:
['ID', 'school1', 'job', 're_registration', 'nationality', 'inflow_route', 'whyBDA', 'what_to_gain', 'hope_for_group', 'desired_career_path', 'project_type', 'time_input', 'incumbents_level', 'incumbents_lecture', 'incumbents_company_level', 'incumbents_lecture_type', 'incumbents_lecture_scale', 'contest_participation', 'completed', 'major_count', 'is_IT', 'class_count', 'previous_class_count', 'incumbents_lecture_scale_reason_len', 'desired_job_count', 'certificate_count', 'desired_certificate_count', 'desired_job_except_data_count', 'onedayclass_topic_count']

Train shape: (748, 29)
Test shape: (814, 28)


In [10]:
# 데이터 타입 변경 (LabelEncoder 전에 수행)
print("=" * 100)
print("데이터 타입 변경")
print("=" * 100)

# int형으로 변환할 컬럼들 (카운트 및 수치형 피처)
int_columns = ['major_count', 'is_IT', 'class_count', 'previous_class_count', 
               'incumbents_lecture_scale_reason_len', 'desired_job_count', 
               'certificate_count', 'desired_certificate_count', 
               'desired_job_except_data_count', 'onedayclass_topic_count', 'time_input']

# int형으로 변환
for col in int_columns:
    df_train[col] = df_train[col].astype(int)
    df_test[col] = df_test[col].astype(int)

print(f"✅ int형으로 변환: {int_columns}")

# category형으로 변환할 컬럼들 (ID, completed, int_columns 제외한 나머지)
all_cols = df_train.columns.tolist()
category_columns = [col for col in all_cols if col not in ['ID', 'completed'] + int_columns]

# category형으로 변환
for col in category_columns:
    df_train[col] = df_train[col].astype('category')
    df_test[col] = df_test[col].astype('category')

print(f"✅ category형으로 변환: {category_columns}")
print("\n" + "=" * 100)
print("데이터 타입 변경 완료!")
print("=" * 100)

데이터 타입 변경
✅ int형으로 변환: ['major_count', 'is_IT', 'class_count', 'previous_class_count', 'incumbents_lecture_scale_reason_len', 'desired_job_count', 'certificate_count', 'desired_certificate_count', 'desired_job_except_data_count', 'onedayclass_topic_count', 'time_input']
✅ category형으로 변환: ['school1', 'job', 're_registration', 'nationality', 'inflow_route', 'whyBDA', 'what_to_gain', 'hope_for_group', 'desired_career_path', 'project_type', 'incumbents_level', 'incumbents_lecture', 'incumbents_company_level', 'incumbents_lecture_type', 'incumbents_lecture_scale', 'contest_participation']

데이터 타입 변경 완료!


In [11]:
df_train

ID school1  job re_registration nationality  \
0    TRAIN_000      22  대학생             아니요         내국인   
1    TRAIN_001       1  대학생             아니요         내국인   
2    TRAIN_002      27  대학생             아니요         내국인   
3    TRAIN_003       1  대학생             아니요         내국인   
4    TRAIN_004      16  대학생             아니요         내국인   
..         ...     ...  ...             ...         ...   
743  TRAIN_743      13  대학생             아니요         내국인   
744  TRAIN_744      43  대학생             아니요         내국인   
745  TRAIN_745      11  대학생             아니요         내국인   
746  TRAIN_746      30  대학생             아니요         내국인   
747  TRAIN_747      25  대학생             아니요         내국인   

                         inflow_route  \
0                               에브리타임   
1                               지인 추천   
2                               인스타그램   
3                               에브리타임   
4                               지인 추천   
..                                ...   
743                             에브리타임   
744  대외활동 사이트(링커리어, 캠퍼스픽, 캠퍼즈, 위비티 등)   
745                             에브리타임   
746                             인스타그램   
747                             에브리타임   

                                                whyBDA what_to_gain  \
0                   큰 규모인 만큼, 커리큘럼이나 운영 등 관리가 잘 될것 같아서      프로젝트 경험   
1                   큰 규모인 만큼, 커리큘럼이나 운영 등 관리가 잘 될것 같아서    데이터 분석 역량   
2    BDA 학회원만의 혜택을 누리고 싶어서(현직자 강연, 잡 페스티벌, 기업연계 공모전 등)       공모전 경험   
3                                         혼자 공부하기 어려워서      프로젝트 경험   
4                   큰 규모인 만큼, 커리큘럼이나 운영 등 관리가 잘 될것 같아서       공모전 경험   
..                                                 ...          ...   
743                                       혼자 공부하기 어려워서       공모전 경험   
744  BDA 학회원만의 혜택을 누리고 싶어서(현직자 강연, 잡 페스티벌, 기업연계 공모전 등)    데이터 분석 역량   
745                                      시간적으로 부담이 없어서      프로젝트 경험   
746                                       혼자 공부하기 어려워서    데이터 분석 역량   
747                                       혼자 공부하기 어려워서    데이터 분석 역량   

               hope_for_group desired_career_path  ... major_count  is_IT  \
0           네. 온라인으로 참여하고 싶어요              대학원 진학  ...           2      0   
1          네. 오프라인으로 참여하고 싶어요                  취업  ...           2      1   
2          네. 오프라인으로 참여하고 싶어요              대학원 진학  ...           1      0   
3           네. 온라인으로 참여하고 싶어요                  취업  ...           2      1   
4          네. 오프라인으로 참여하고 싶어요                  취업  ...           2      1   
..                        ...                 ...  ...         ...    ...   
743        네. 오프라인으로 참여하고 싶어요                  취업  ...           2      1   
744         네. 온라인으로 참여하고 싶어요                  취업  ...           1      0   
745        네. 오프라인으로 참여하고 싶어요              대학원 진학  ...           2      0   
746  아니요. 개인적으로 학회 활동을 하고 싶어요                  취업  ...           1      1   
747        네. 오프라인으로 참여하고 싶어요                  취업  ...           1      0   

    class_count previous_class_count incumbents_lecture_scale_reason_len  \
0             2                    0                                  28   
1             1                    0                                  48   
2             1                    0                                  57   
3             1                    0                                  25   
4             1                    0                                  24   
..          ...                  ...                                 ...   
743           1                    0                                  35   
744           1                    0                                  19   
745           1                    0                                  22   
746           1                    0                                   6   
747           1                    0                                   4   

    desired_job_count certificate_count desired_certificate_count  \
0                   1                 1           

In [12]:
# 나머지 범주형 변수들 LabelEncoder 적용 (completed 제외)
from sklearn.preprocessing import LabelEncoder

# 범주형 컬럼 리스트 (object 타입 + category 타입)
categorical_cols = df_train.select_dtypes(include=['object', 'category']).columns.tolist()

# ID, completed 제외 (completed는 타겟 변수, ID는 식별자)
if 'ID' in categorical_cols:
    categorical_cols.remove('ID')
if 'completed' in categorical_cols:
    categorical_cols.remove('completed')

print("=" * 100)
print("LabelEncoder 적용")
print("=" * 100)
print(f"대상 컬럼: {categorical_cols}\n")

# LabelEncoder 딕셔너리
label_encoders = {}

for col in categorical_cols:
    le = LabelEncoder()
    
    # Train과 Test 데이터 합쳐서 fit (동일한 인코딩 규칙 적용)
    combined = pd.concat([df_train[col], df_test[col]])
    le.fit(combined)
    
    # Transform
    df_train[col] = le.transform(df_train[col])
    df_test[col] = le.transform(df_test[col])
    
    label_encoders[col] = le
    
    print(f"✅ {col}: {len(le.classes_)} classes")

print("\n" + "=" * 100)
print("LabelEncoder 완료!")
print("=" * 100)

LabelEncoder 적용
대상 컬럼: ['school1', 'job', 're_registration', 'nationality', 'inflow_route', 'whyBDA', 'what_to_gain', 'hope_for_group', 'desired_career_path', 'project_type', 'incumbents_level', 'incumbents_lecture', 'incumbents_company_level', 'incumbents_lecture_type', 'incumbents_lecture_scale', 'contest_participation']

✅ school1: 99 classes
✅ job: 4 classes
✅ re_registration: 2 classes
✅ nationality: 3 classes
✅ inflow_route: 14 classes
✅ whyBDA: 10 classes
✅ what_to_gain: 12 classes
✅ hope_for_group: 3 classes
✅ desired_career_path: 5 classes
✅ project_type: 2 classes
✅ incumbents_level: 2 classes
✅ incumbents_lecture: 11 classes
✅ incumbents_company_level: 26 classes
✅ incumbents_lecture_type: 3 classes
✅ incumbents_lecture_scale: 10 classes
✅ contest_participation: 7 classes

LabelEncoder 완료!


In [13]:
df_train.head()

ID  school1  job  re_registration  nationality  inflow_route  \
0  TRAIN_000       22    0                0            0             9   
1  TRAIN_001        1    0                0            0            12   
2  TRAIN_002       27    0                0            0            10   
3  TRAIN_003        1    0                0            0             9   
4  TRAIN_004       16    0                0            0            12   

   whyBDA  what_to_gain  hope_for_group  desired_career_path  ...  \
0       4            11               1                    0  ...   
1       4             4               0                    3  ...   
2       0             1               0                    0  ...   
3       9            11               1                    3  ...   
4       4             1               0                    3  ...   

   major_count  is_IT  class_count  previous_class_count  \
0            2      0            2                     0   
1            2      1            1                     0   
2            1      0            1                     0   
3            2      1            1                     0   
4            2      1            1                     0   

   incumbents_lecture_scale_reason_len  desired_job_count  certificate_count  \
0                                   28                  1                  1   
1                                   48                  2                  0   
2                                   57                  3                  1   
3                                   25                  2                  2   
4                                   24                  1                  1   

   desired_certificate_count  desired_job_except_data_count  \
0                          5                              1   
1                          4                              2   
2                          3                              3   
3                          2                              1   
4                          4                              3   

   onedayclass_topic_count  
0                        5  
1                        1  
2                        3  
3                        1  
4                        3  

[5 rows x 29 columns]

In [14]:
# is_IT를 0, 1로 변환 (이미 int형이지만 명시적으로)
df_train['is_IT'] = df_train['is_IT'].astype(int)
df_test['is_IT'] = df_test['is_IT'].astype(int)

print("✅ is_IT를 0, 1로 변환 완료")

# 열 이름 변경: ID, completed 제외하고 F1 ~ F32로 변경
cols = df_train.columns.tolist()

# ID, completed 제외한 피처 컬럼
feature_cols = [c for c in cols if c not in ['ID', 'completed']]

# 새로운 컬럼명 매핑 생성
column_mapping = {feature_cols[i]: f'F{i+1}' for i in range(len(feature_cols))}

# 컬럼명 변경
df_train = df_train.rename(columns=column_mapping)
df_test = df_test.rename(columns=column_mapping)

print(f"✅ 열 이름 변경 완료: F1 ~ F{len(feature_cols)}")

# completed를 y로 이름 변경하고 맨 마지막으로 이동
df_train = df_train.rename(columns={'completed': 'y'})

# 컬럼 순서 재정렬: ID, F1~F32, y
cols = df_train.columns.tolist()
cols.remove('y')
df_train = df_train[cols + ['y']]

print("✅ completed → y 로 변경 및 맨 마지막으로 이동")
print(f"최종 컬럼 순서: {df_train.columns.tolist()}")

✅ is_IT를 0, 1로 변환 완료
✅ 열 이름 변경 완료: F1 ~ F27
✅ completed → y 로 변경 및 맨 마지막으로 이동
최종 컬럼 순서: ['ID', 'F1', 'F2', 'F3', 'F4', 'F5', 'F6', 'F7', 'F8', 'F9', 'F10', 'F11', 'F12', 'F13', 'F14', 'F15', 'F16', 'F17', 'F18', 'F19', 'F20', 'F21', 'F22', 'F23', 'F24', 'F25', 'F26', 'F27', 'y']


In [15]:
# 데이터 타입 최종 설정 (F1~F21: category, F22~F31: int)
print("=" * 100)
print("최종 데이터 타입 설정")
print("=" * 100)

# int형으로 유지할 컬럼 (F22~F31: 카운트 피처)
int_feature_list = [f'F{i}' for i in range(22, 32)]

# category형으로 변환할 컬럼 (F1~F21: 인코딩된 범주형 피처)
category_feature_list = [f'F{i}' for i in range(1, 22)]

# category형으로 변환
for col in category_feature_list:
    df_train[col] = df_train[col].astype('category')
    df_test[col] = df_test[col].astype('category')

print(f"✅ int형 컬럼 ({len(int_feature_list)}개): {int_feature_list}")
print(f"✅ category형 컬럼 ({len(category_feature_list)}개): {category_feature_list}")
print("\n" + "=" * 100)
print("최종 데이터 타입 설정 완료!")
print("=" * 100)

최종 데이터 타입 설정
✅ int형 컬럼 (10개): ['F22', 'F23', 'F24', 'F25', 'F26', 'F27', 'F28', 'F29', 'F30', 'F31']
✅ category형 컬럼 (21개): ['F1', 'F2', 'F3', 'F4', 'F5', 'F6', 'F7', 'F8', 'F9', 'F10', 'F11', 'F12', 'F13', 'F14', 'F15', 'F16', 'F17', 'F18', 'F19', 'F20', 'F21']

최종 데이터 타입 설정 완료!


In [16]:
df_train

ID  F1 F2 F3 F4  F5 F6  F7 F8 F9  ... F19 F20 F21 F22 F23 F24 F25  \
0    TRAIN_000  22  0  0  0   9  4  11  1  0  ...   0   2   0  28   1   1   5   
1    TRAIN_001   1  0  0  0  12  4   4  0  3  ...   1   1   0  48   2   0   4   
2    TRAIN_002  27  0  0  0  10  0   1  0  0  ...   0   1   0  57   3   1   3   
3    TRAIN_003   1  0  0  0   9  9  11  1  3  ...   1   1   0  25   2   2   2   
4    TRAIN_004  16  0  0  0  12  4   1  0  3  ...   1   1   0  24   1   1   4   
..         ...  .. .. .. ..  .. ..  .. .. ..  ...  ..  ..  ..  ..  ..  ..  ..   
743  TRAIN_743  13  0  0  0   9  9   1  0  3  ...   1   1   0  35   1   2   2   
744  TRAIN_744  43  0  0  0   8  0   4  1  3  ...   0   1   0  19   1   2   1   
745  TRAIN_745  11  0  0  0   9  2  11  0  0  ...   0   1   0  22   1   0   2   
746  TRAIN_746  30  0  0  0  10  9   4  2  3  ...   1   1   0   6   2   0   2   
747  TRAIN_747  25  0  0  0   9  9   4  0  3  ...   0   1   0   4   1   1   1   

    F26 F27  y  
0     1   5  0  
1     2   1  0  
2     3   3  0  
3     1   1  1  
4     3   3  0  
..   ..  .. ..  
743   2   4  0  
744   2   6  1  
745   1   1  1  
746   3   3  1  
747   1   1  1  

[748 rows x 29 columns]

In [17]:
df_test

ID  F1 F2 F3 F4  F5 F6  F7 F8 F9  ... F18 F19 F20 F21 F22 F23 F24  \
0    TEST_000   3  0  0  0  12  2   4  2  0  ...   2   0   1   0  38   1   0   
1    TEST_001  45  0  0  0   9  4   4  1  3  ...   2   0   1   0  26   1   0   
2    TEST_002  56  0  0  0   9  9   4  1  3  ...   2   0   1   0  13   1   0   
3    TEST_003  48  0  0  0   9  9  11  0  3  ...   2   0   2   0  21   1   4   
4    TEST_004   0  0  0  0  10  4   4  1  3  ...   0   0   1   0  15   1   0   
..        ...  .. .. .. ..  .. ..  .. .. ..  ...  ..  ..  ..  ..  ..  ..  ..   
809  TEST_809  30  0  0  0   9  9   4  0  3  ...   2   0   1   0  22   1   0   
810  TEST_810  22  0  0  0   9  9  11  1  3  ...   2   0   1   0  39   1   0   
811  TEST_811  20  0  0  0   9  4   4  0  0  ...   2   0   1   0  16   1   0   
812  TEST_812  15  2  0  0   9  9   9  0  3  ...   2   0   1   0  25   1   1   
813  TEST_813   0  0  0  0  10  8   4  1  3  ...   0   0   1   0   3   1   0   

    F25 F26 F27  
0     6   1   1  
1     2   1   1  
2     3   3   1  
3     2   5   1  
4     3   1   4  
..   ..  ..  ..  
809   3   1   5  
810   2   4   2  
811   2   1   1  
812   4   1   2  
813   3   1   4  

[814 rows x 28 columns]

In [18]:
import os

# 저장 경로 설정
save_path = r"C:\Users\solba\dacon-project\data\preprocessing"

# 디렉토리가 없으면 생성
os.makedirs(save_path, exist_ok=True)

# 전처리된 데이터 저장
df_train.to_csv(os.path.join(save_path, "train_preprocessed.csv"), index=False)
df_test.to_csv(os.path.join(save_path, "test_preprocessed.csv"), index=False)

print("=" * 100)
print("데이터 저장 완료!")
print("=" * 100)
print(f"저장 경로: {save_path}")
print(f"Train 파일: train_preprocessed.csv ({df_train.shape})")
print(f"Test 파일: test_preprocessed.csv ({df_test.shape})")

데이터 저장 완료!
저장 경로: C:\Users\solba\dacon-project\data\preprocessing
Train 파일: train_preprocessed.csv ((748, 29))
Test 파일: test_preprocessed.csv ((814, 28))


In [19]:
# 데이터 타입별 컬럼 확인
print("=" * 100)
print("데이터 타입별 컬럼 정보")
print("=" * 100)

# 역매핑: F 컬럼 -> 원래 컬럼명
reverse_mapping = {v: k for k, v in column_mapping.items()}

# int형 컬럼 확인
int_feature_cols = [col for col in df_train.columns if col.startswith('F') and df_train[col].dtype == 'int64']
print(f"\n✅ int형 컬럼 ({len(int_feature_cols)}개):")
for col in sorted(int_feature_cols, key=lambda x: int(x[1:])):
    original_name = reverse_mapping.get(col, "Unknown")
    print(f"   {col}: {original_name}")

# category형 컬럼 확인  
category_feature_cols = [col for col in df_train.columns if col.startswith('F') and df_train[col].dtype != 'int64']
print(f"\n✅ category형 컬럼 ({len(category_feature_cols)}개):")
for col in sorted(category_feature_cols, key=lambda x: int(x[1:])):
    original_name = reverse_mapping.get(col, "Unknown")
    print(f"   {col}: {original_name}")

print("\n" + "=" * 100)

데이터 타입별 컬럼 정보

✅ int형 컬럼 (6개):
   F22: incumbents_lecture_scale_reason_len
   F23: desired_job_count
   F24: certificate_count
   F25: desired_certificate_count
   F26: desired_job_except_data_count
   F27: onedayclass_topic_count

✅ category형 컬럼 (21개):
   F1: school1
   F2: job
   F3: re_registration
   F4: nationality
   F5: inflow_route
   F6: whyBDA
   F7: what_to_gain
   F8: hope_for_group
   F9: desired_career_path
   F10: project_type
   F11: time_input
   F12: incumbents_level
   F13: incumbents_lecture
   F14: incumbents_company_level
   F15: incumbents_lecture_type
   F16: incumbents_lecture_scale
   F17: contest_participation
   F18: major_count
   F19: is_IT
   F20: class_count
   F21: previous_class_count

